### PR TITLE
Create new anonymizer for new engine

### PR DESCRIPTION
**Pull request by matthewryanwells**
### Description
Created a new listener for the SQLSyntaxParser to anonymize all queries going through the new engine, calling the old anonymizer only if the query falls back to the legacy engine.

Through the update of the anonymizer there are a few differences between between the current and old anonymization including:

- Dots are used to access elements of a table, e.g. `identifier.identifier` instead of simplifying as just `identifier`
- All functions are capitalized
- Adds a space between all nodes except for before and after a dot, before a comma, and between compound comparison symbols (<=, >=, <>, !=)
- Includes the negative symbol when anonymizing numbers, e.g. `number / - number` instead `number / number`
 
### Issues Resolved
https://github.com/opensearch-project/sql/issues/821
 
### Check List
- [x] New functionality includes testing.
  - [x] All tests pass, including unit test, integration test and doctest
- [x] New functionality has been documented.
  - [x] New functionality has javadoc added
  - [x] New functionality has user manual doc added
- [x] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Review by MaxKsyunz — DISMISSED** [review]

File: sql/src/main/java/org/opensearch/sql/sql/antlr/SQLSyntaxParser.java [hunk 1]
```diff
@@ -26,10 +34,16 @@ public class SQLSyntaxParser implements Parser {
    */
   @Override
   public ParseTree parse(String query) {
+    AnonymizerListener anonymizer = new AnonymizerListener();
+
     OpenSearchSQLLexer lexer = new OpenSearchSQLLexer(new CaseInsensitiveCharStream(query));
     OpenSearchSQLParser parser = new OpenSearchSQLParser(new CommonTokenStream(lexer));
-    parser.addErrorListener(new SyntaxAnalysisErrorListener());
```
Comment (acarbonetto) [inline sql/src/main/java/org/opensearch/sql/sql/antlr/SQLSyntaxParser.java:31]: We should revert this line (31).  There's no reason to update this anymore.

File: sql/src/test/java/org/opensearch/sql/sql/parser/AnonymizerListenerTest.java [hunk 2]
```diff
@@ -0,0 +1,211 @@
...
+
+  public AnonymizerListener anonymizerListener = new AnonymizerListener();
+
+  public SyntaxAnalysisErrorListener errorListener = new SyntaxAnalysisErrorListener();
+
+  /**
+   * Helper function to parse SQl queries for testing purposes.
+   * @param query SQL query to be anonymized.
+   */
+  public void parse(String query) {
```
Comment (acarbonetto) [inline sql/src/test/java/org/opensearch/sql/sql/parser/AnonymizerListenerTest.java:31]: private

**Review by acarbonetto — APPROVED** [review]

**Review by MaxKsyunz — APPROVED** [review]

File: sql/src/main/java/org/opensearch/sql/sql/antlr/AnonymizerListener.java [hunk 3]
```diff
@@ -0,0 +1,107 @@
+package org.opensearch.sql.sql.antlr;
```
Comment (Yury-Fridlyand) [inline sql/src/main/java/org/opensearch/sql/sql/antlr/AnonymizerListener.java:1]: License header is missing

Comment (Yury-Fridlyand) [reply sql/src/main/java/org/opensearch/sql/sql/antlr/AnonymizerListener.java:1]: All rest LGFM

**Review by acarbonetto — APPROVED** [review]

**Review by Yury-Fridlyand — APPROVED** [review]

**Review by Yury-Fridlyand — APPROVED** [review]

**Pull request merged**